query-node/examples/kusama-query-node/mappings/treasury/proposal.ts

```diff
@@ -0,0 +1,46 @@
+import { SubstrateEvent, DB } from '../../generated/indexer';
+import { Proposal } from '../../generated/graphql-server/src/modules/proposal/proposal.model';
+import { ProposalStatus } from '../../generated/graphql-server/src/modules/enums/enums';
+import { assert } from 'console';
+
+// New proposal
+export async function handleProposed(db: DB, event: SubstrateEvent) {
+  const { ProposalIndex } = event.event_params;
+  if (event.extrinsic) {
+    const proposal = new Proposal();
+    proposal.proposalIndex = ProposalIndex.toString();
```

Comment:
Seeing this mapping made me make this issue https://github.com/Joystream/joystream/issues/1024 for the future.